src/components/text-editor/prosemirror-adapter/plugins/trigger/inserter.ts

```diff
@@ -9,36 +9,59 @@ export const inserterFactory = (view: EditorView): TextEditor => {
     const startPos = getTriggerStartPosition(view);
 
     return {
-        insert: (input: TextEditorNode | string) => {
-            const { state, dispatch } = view;
-            const { schema, selection } = state;
-            const { $from } = selection;
+        insert: insert(view, startPos),
+        insertHtml: insertHtml(view, startPos),
```

Comment:
These should be called factory or something like it. It looks a bit confusing now with the curried functions. I don't have a better imagination than the proposals below.

```suggestion
        insert: createNodeAndTextInserter(view, startPos),
        insertHtml: createHtmlInserter(view, startPos),
```